packages/pretty-format/src/index.js

```diff
@@ -41,6 +41,29 @@ const errorToString = Error.prototype.toString;
 const regExpToString = RegExp.prototype.toString;
 const symbolToString = Symbol.prototype.toString;
 
+// Return whether val is equal to global window object.
+let noWindow;
+let theWindow;
+const isWindow = val => {
```

Comment:
I still don't think this is necessary. This change passes your test as well:

```diff
diff --git i/packages/pretty-format/src/index.js w/packages/pretty-format/src/index.js
index 35a591c9..ba4f91f1 100644
--- i/packages/pretty-format/src/index.js
+++ w/packages/pretty-format/src/index.js
@@ -41,6 +41,8 @@ const errorToString = Error.prototype.toString;
 const regExpToString = RegExp.prototype.toString;
 const symbolToString = Symbol.prototype.toString;
 
+const isWindow = val => typeof window !== 'undefined' && val === window;
+
 const SYMBOL_REGEXP = /^Symbol\((.*)\)(.*)$/;
 const NEWLINE_REGEXP = /\n/gi;
 
@@ -224,7 +226,7 @@ function printComplexValue(
         '}';
   }
 
-  return hitMaxDepth
+  return hitMaxDepth || isWindow(val)
     ? '[' + (val.constructor ? val.constructor.name : 'Object') + ']'
     : (min ? '' : (val.constructor ? val.constructor.name : 'Object') + ' ') +
       '{' +
```

Reply:
I suppose you can cache the `typeof` check, but that doesn't seem like an optimization

Reply:
Yes indeed, `typeof window` works in my experimental codes where `window` throws.

Thank you very much, I will update pull request according to your one liner above.